compiler/rustc_typeck/src/check/writeback.rs

```diff
@@ -384,9 +384,9 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
         assert_eq!(fcx_typeck_results.hir_owner, self.typeck_results.hir_owner);
         let common_hir_owner = fcx_typeck_results.hir_owner;
 
-        for (&id, &origin) in fcx_typeck_results.closure_kind_origins().iter() {
-            let hir_id = hir::HirId { owner: common_hir_owner, local_id: id };
-            self.typeck_results.closure_kind_origins_mut().insert(hir_id, origin);
+        for (id, origin) in fcx_typeck_results.closure_kind_origins().iter() {
+            let hir_id = hir::HirId { owner: common_hir_owner, local_id: *id };
+            self.typeck_results.closure_kind_origins_mut().insert(hir_id, origin.clone());
```

Comment:
So the `Place` might contain types that have been resolved but we just didn't update their reference with the `Place`. Also, all the regions used in references need to be changed to `ty::ReErased` or something similar. This should probably do it:

```suggestion
            let place_span = origin.1;
            let place = self.resolve(origin.0.clone(), place_span);

            self.typeck_results.closure_kind_origins_mut().insert(hir_id, (place, *place_span));
```